game/ffmpeg.cc

```diff
@@ -183,18 +191,39 @@ void FFmpeg::seek_internal() {
 void FFmpeg::decodePacket() {
 #if LIBAVCODEC_VERSION_INT >= (AV_VERSION_INT(57, 37, 0))
 	AVPacket pkt;
-	while (av_read_frame(m_formatContext, &pkt) >= 0) {
+	while (true) {
+		int ret = av_read_frame(m_formatContext, &pkt);
+		if(ret == AVERROR_EOF) {
+			// End of file: no more data to read.
+			throw FFmpeg::eof_error();
+		} else if(ret < 0) {
+			// FIXME: we might want to return an error here.
+			break;
+		}
 		if (m_quit || m_seekTarget == m_seekTarget) return; // something weird required
 		if (pkt.stream_index != m_streamId) return; // wrong stream
-		int ret = avcodec_send_packet(m_codecContext, &pkt);
-		if (ret < 0 || ret == AVERROR(EAGAIN) || ret == AVERROR_EOF) {
-			// error
+		ret = avcodec_send_packet(m_codecContext, &pkt);
+		if(ret == AVERROR_EOF) {
+			// End of file: no more data to read.
+			throw FFmpeg::eof_error();
+		} else if(ret == AVERROR(EAGAIN)) {
+			// not enough data for decoder, read more
+			continue;
+		} else if(ret < 0) {
+			// FIXME: we might want to return an error here.
```

Comment:
Either throw the error, or remove the comment. OR remove the comment, and add a log line with the error.

Reply:
now throw